src/Gandi.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Nexylan packages.
+ *
+ * (c) Nexylan SAS <contact@nexylan.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Nexy\Gandi;
+
+use fXmlRpc\Client as FxmlrpcClient;
+use fXmlRpc\Proxy;
+use fXmlRpc\Transport\HttpAdapterTransport;
+use Http\Adapter\Guzzle6\Client as GuzzleClient;
+use Http\Client\HttpClient;
+use Http\Discovery\HttpClientDiscovery;
+use Http\Message\MessageFactory\GuzzleMessageFactory;
+use Nexy\Gandi\Api\AbstractApi;
+
+/**
+ * @author Jérôme Pogeant <p-jerome@hotmail.fr>
+ */
+final class Gandi
+{
+    /**
+     * @var HttpClient
+     */
+    private $httpClient;
+
+    /**
+     * @var string
+     */
+    private $apiUrl;
+
+    /**
+     * @var string
+     */
+    private $apiKey;
+
+    /**
+     * @param HttpClient $httpClient
+     * @param string     $apiUrl
+     * @param string     $apiKey
+     */
+    public function __construct(HttpClient $httpClient, string $apiUrl, string $apiKey)
+    {
+        $this->httpClient = $httpClient ?: HttpClientDiscovery::find();
+        $this->apiUrl = $apiUrl;
+        $this->apiKey = $apiKey;
+    }
+
+    /**
+     * @param string $name
+     * @param array  $arguments
+     *
+     * @return AbstractApi
+     */
+    public function __call(string $name, array $arguments): AbstractApi
+    {
+        try {
+            return $this->api(ucfirst(str_replace('api', '', $name)));
+        } catch (\InvalidArgumentException $e) {
+            throw new \BadMethodCallException(sprintf('Undefined method %s', $name));
+        }
+    }
+
+    /**
+     * Return Gandi proxy.
+     *
+     * @return Proxy
+     */
+    public function setup(): Proxy
```

Comment:
This is not needed. Just instanciate a new `FxmlrpcClient` with possibility to give an another one an set sur url.

Example (might not work as is):

```php
    /**
     * @var Proxy
     */
    private $client;

    public function __construct(string $apiUrl, string $apiKey, FxmlrpcClient $xmlClient = null)
    {
        $xmlClient = $xmlClient ?: new FxmlrpcClient();
        $xmlClient->setUri($apiUrl);
        $xmlClient->prependParams([$apiKey]);
        $this->client = new Proxy($xmlClient);
    }
```